File: edukate-gateway/src/main/java/io/github/sanyavertolet/edukate/gateway/dtos/SignUpRequest.java
package io.github.sanyavertolet.edukate.gateway.dtos;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
public class SignUpRequest {
    @NotBlank
    private String username;

    @ToString.Exclude
    @NotBlank
    private String password;

    @Email
    @NotBlank
    private String email;
}

File: edukate-gateway/src/main/kotlin/io/github/sanyavertolet/edukate/gateway/dtos/SignUpRequest.kt
package io.github.sanyavertolet.edukate.gateway.dtos

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size

data class SignUpRequest(
    @field:NotBlank
    @field:Size(min = 3, max = 15)
    @field:Pattern(regexp = "^[a-zA-Z][a-zA-Z0-9_-]+[a-zA-Z0-9]$")
    val username: String,
    @field:NotBlank @field:Size(min = 6, max = 128) val password: String,
    @field:NotBlank @field:Email val email: String,
) {
    override fun toString(): String = "SignUpRequest(username=$username, email=$email)"
}
